registry/storage/catalog.go

```
@@ -93,5 +99,38 @@ func (reg *registry) Enumerate(ctx context.Context, ingester func(string) error)
 		}
 	}
 	return nil
+}
+
+// todo(richardscothern): find a better place for this
+func (reg *metadataRegistry) Repositories(ctx context.Context, repos []string, lastRepo string) (n int, errVal error) {
+	if len(repos) == 0 {
+		return 0, errors.New("no space in slice")
```

What is the need to return this error rather than just `return 0, nil`